pkg/config/user_config_validation.go

```diff
@@ -25,3 +31,41 @@ func validateEnum(name string, value string, allowedValues []string) error {
 	allowedValuesStr := strings.Join(allowedValues, ", ")
 	return fmt.Errorf("Unexpected value '%s' for '%s'. Allowed values: %s", value, name, allowedValuesStr)
 }
+
+func validateKeybindingsRecurse(path string, node any) error {
+	value := reflect.ValueOf(node)
+	if value.Kind() == reflect.Struct {
+		for _, field := range reflect.VisibleFields(reflect.TypeOf(node)) {
+			var newPath string
+			if len(path) == 0 {
+				newPath = field.Name
+			} else {
+				newPath = fmt.Sprintf("%s.%s", path, field.Name)
+			}
+			if err := validateKeybindingsRecurse(newPath,
+				value.FieldByName(field.Name).Interface()); err != nil {
+				return err
+			}
+		}
+	} else if value.Kind() == reflect.Slice {
+		for i := 0; i < value.Len(); i++ {
+			if err := validateKeybindingsRecurse(
+				fmt.Sprintf("%s[%d]", path, i), value.Index(i).Interface()); err != nil {
+				return err
+			}
+		}
+	} else if value.Kind() == reflect.String {
+		key := node.(string)
+		if !isValidKeybindingKey(key) {
+			return fmt.Errorf("Unrecognized key '%s' for keybinding '%s'. For permitted values see %s",
+				strings.ToLower(key), path, constants.Links.Docs.CustomKeybindings)
```

Comment:
Unfortunately we can't i18n-ize these, because the config is read very early, before the translation system is initialized. And it is important that we validate the config right after loading, so that all clients of it can just use it without having to worry about valid data.